pinecone/client_test.go

```diff
@@ -317,200 +108,65 @@ func (ts *IntegrationTests) TestDescribeNonExistentIndex() {
-	}
-
 	// Call the method under test to list all collections
 	collections, err := ts.client.ListCollections(ctx)
 	require.NoError(ts.T(), err)
-	require.Greater(ts.T(), len(collections), 2, "Expected at least three collections to exist")
+	require.Greater(ts.T(), len(collections), 0, "Expected at least one collection to exist")
 
-	// Check that the created collections are in the returned list
-	found := 0
+	// Check that the created collection is returned in the list
+	found := false
 	for _, collection := range collections {
-		for _, name := range collectionNames {
-			if collection.Name == name {
-				found++
-				break
-			}
+		if collection.Name == ts.collectionName {
+			found = true
 		}
 	}
-	require.Equal(ts.T(), len(collectionNames), found, "Not all created collections were listed")
+	require.True(ts.T(), found, "Collection %v not found in list of collections", ts.collectionName)
 }
 
 func (ts *IntegrationTests) TestDescribeCollection() {
-	if ts.indexType == "serverless" {
-		ts.T().Skip("No pod index to test")
-	}
 	ctx := context.Background()
-	collectionName := uuid.New().String()
-
-	defer func(client *Client, ctx context.Context, collectionName string) {
-		err := client.DeleteCollection(ctx, collectionName)
-		require.NoError(ts.T(), err)
-	}(ts.client, ctx, collectionName)
-
-	_, err := ts.client.CreateCollection(ctx, &CreateCollectionRequest{
-		Name:   collectionName,
-		Source: ts.idxName,
-	})
-	require.NoError(ts.T(), err)
-
-	collection, err := ts.client.DescribeCollection(ctx, collectionName)
-	require.NoError(ts.T(), err)
-	require.Equal(ts.T(), collectionName, collection.Name, "Collection name does not match")
-}
-
-func (ts *IntegrationTests) TestCreateCollection() {
-	if ts.indexType == "serverless" {
-		ts.T().Skip("No pod index to test")
-	}
-	name := uuid.New().String()
-	sourceIndex := ts.idxName
-
-	defer func() {
-		err := ts.client.DeleteCollection(context.Background(), name)
-		require.NoError(ts.T(), err)
-	}()
-
-	collection, err := ts.client.CreateCollection(context.Background(), &CreateCollectionRequest{
-		Name:   name,
-		Source: sourceIndex,
-	})
-
-	require.NoError(ts.T(), err)
-	require.Equal(ts.T(), name, collection.Name, "Collection name does not match")
-}
-
-func (ts *IntegrationTests) TestDeleteCollection() {
 	if ts.indexType == "serverless" {
 		ts.T().Skip("No pod index to test")
 	}
-	collectionName := uuid.New().String()
-	_, err := ts.client.CreateCollection(context.Background(), &CreateCollectionRequest{
-		Name:   collectionName,
-		Source: ts.idxName,
-	})
-
-	require.NoError(ts.T(), err)
 
-	err = ts.client.DeleteCollection(context.Background(), collectionName)
+	collection, err := ts.client.DescribeCollection(ctx, ts.collectionName)
 	require.NoError(ts.T(), err)
+	require.Equal(ts.T(), ts.collectionName, collection.Name, "Collection name does not match")
 }
 
-func (ts *IntegrationTests) TestDeletionProtectionPodIntegration() {
```

Comment:
There used to be separate tests for this for pod vs. serverless which we don't need, we can handle both suites in one test.